frontend/widgets/OBSBasic_SceneCollections.cpp

```diff
@@ -841,60 +843,78 @@ static obs_data_t *GenerateSaveData(obs_data_array_t *sceneOrder, obs_data_array
 		},
 		static_cast<void *>(&FilterAudioSources));
 
-	/* -------------------------------- */
-	/* save group sources separately    */
+	// We're saving groups separately ensures they won't be loaded in older versions.
+	// ToDo: Get rid of this at some point. Groups were introduced in 22.0
+	OBSDataArrayAutoRelease groupsArray;
 
-	/* saving separately ensures they won't be loaded in older versions */
-	OBSDataArrayAutoRelease groupsArray = obs_save_sources_filtered(
-		[](void *, obs_source_t *source) { return obs_source_is_group(source); }, nullptr);
+	auto sourcesAndGroups = std::make_pair(sourcesArray.Get(), groupsArray.Get());
+	using sourcesAndGroups_t = decltype(sourcesAndGroups);
 
-	/* -------------------------------- */
+	auto sceneSaveCallback = [](void *param, obs_source_t *source) -> bool {
+		auto sourcesArrays = static_cast<sourcesAndGroups_t *>(param);
+		obs_data_t *source_data = obs_save_source(source);
 
-	OBSSourceAutoRelease transition = obs_get_output_source(0);
-	obs_source_t *currentScene = obs_scene_get_source(scene);
-	const char *sceneName = obs_source_get_name(currentScene);
-	const char *programName = obs_source_get_name(curProgramScene);
+		if (obs_source_is_scene(source)) {
+			obs_data_array_push_back(sourcesArrays->first, source_data);
+		} else {
+			obs_data_array_push_back(sourcesArrays->second, source_data);
+		}
+		return true;
+	};
 
-	const char *sceneCollection = config_get_string(App()->GetUserConfig(), "Basic", "SceneCollection");
+	obs_enum_scenes(sceneSaveCallback, &sourcesAndGroups);
 
-	obs_data_set_string(saveData, "current_scene", sceneName);
-	obs_data_set_string(saveData, "current_program_scene", programName);
-	obs_data_set_array(saveData, "scene_order", sceneOrder);
-	obs_data_set_string(saveData, "name", sceneCollection);
-	obs_data_set_array(saveData, "sources", sourcesArray.Get());
-	obs_data_set_array(saveData, "groups", groupsArray.Get());
-	obs_data_set_array(saveData, "quick_transitions", quickTransitionData);
-	obs_data_set_array(saveData, "transitions", transitions);
-	obs_data_set_array(saveData, "saved_projectors", savedProjectorList);
-	obs_data_set_array(saveData, "canvases", savedCanvases);
+	// Iterate over our additional canvases (if any), save their scenes and groups.
```

Comment:
Why is there a distinction in code between the "main" canvas and "additional canvases"? Wouldn't it be simpler to iterate over all canvases and save them (if they are eligible for saving), as that would include the main canvas by default anyway?